File: #04/q13 - quadperf/quadperf.js
quadperf = x => checar(1, x)

checar = (a, b) => a == b? false
                   : (a * a == b) ? true
                     : checar(a+1, b)

console.log(quadperf(25))
console.log(quadperf(12))
console.log(quadperf(16))
console.log(quadperf(5))

File: #04/q13 - quadperf/quadperf.hs

quadperf :: (Eq t, Num t) => t -> Bool
quadperf a = checar 1 a

checar :: (Eq t, Num t) => t -> t -> Bool
checar x y
  | x == y = False
  | x * x == y = True
  | otherwise = checar (x+1) y


main :: IO ()
main = do
    a <- readLn :: IO Int
    print $ quadperf a